### PR TITLE
fix: Fixed cluster::guild_create, cluster::stage_instance_create and cluster::sticker_packs_get returning 404 Not Found

### PR DESCRIPTION
**Pull request by Commandserver**
- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

The post_rest methods did append a `/` when no major_parameters was passed. This let `cluster::guild_create`, `cluster::stage_instance_create` and `cluster::sticker_packs_get` return 404 Not found because the path was wrong with the appended slash!

The `cluster::sticker_packs_get` method needed an extra fix too.

### CODE REVIEW

File: src/dpp/cluster.cpp [hunk 1]
```diff
@@ -300,7 +300,7 @@ json error_response(const std::string& message, http_request_completion_t& rv)
 
 void cluster::post_rest(const std::string &endpoint, const std::string &major_parameters, const std::string &parameters, http_method method, const std::string &postdata, json_encode_t callback, const std::string &filename, const std::string &filecontent, const std::string &filemimetype) {
 	/* NOTE: This is not a memory leak! The request_queue will free the http_request once it reaches the end of its lifecycle */
-	rest->post_request(new http_request(endpoint + "/" + major_parameters, parameters, [endpoint, callback](http_request_completion_t rv) {
+	rest->post_request(new http_request(endpoint + (!major_parameters.empty() ? "/" : "") + major_parameters, parameters, [endpoint, callback](http_request_completion_t rv) {
```
Comment (Mishura4) [inline src/dpp/cluster.cpp:303]:
```diff
- (!major_parameters.empty() ? "/" : "") + major_parameters
+ (!major_parameters.empty() ? "/" + major_parameters : "")
```
Would be slightly better I think! Not super important though.
(Same for the other one below)

**Comment by Mishura4** [comment]
You also haven't checked "I have ensured that I did not break any existing API calls." in the PR checklist, I'm assuming you're testing it now? I'll approve it when you check it

File: src/dpp/cluster.cpp (hunk 1 above)
Comment (Commandserver) [reply src/dpp/cluster.cpp:303]: hm. not a fan of longer ternary operators. they make it harder to read if you ask me

**Comment by Commandserver** [comment]
> You also haven't checked "I have ensured that I did not break any existing API calls." in the PR checklist, I'm assuming you're testing it now? I'll approve it when you check it

tbh i would have to check almost **all** cluster methods to make sure nothing breaks. I won't do that lol. i can test a few manually if you want. i thought the unit tests cover some cluster methods and that may be enough 🤷.

**Comment by Mishura4** [comment]
Yeah as long as you ran the unit tests and checked the ones you've changed that's fine I think!

**Comment by Mishura4** [comment]
(Just make sure you do check it to make openssf and the other things happy hehe)

**Comment by Commandserver** [comment]
> Yeah as long as you ran the unit tests and checked the ones you've changed that's fine I think!

The ones this pr fixes aren't covered by the unit tests iirc. But i tested them manually of course.

**Comment by Commandserver** [comment]
> (Just make sure you do check the box to make openssf and the other things happy hehe)

okay 😎

**Review by Mishura4 — APPROVED** [review]

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**